### PR TITLE
Rest initialization with mongoDB and basic implementation

### PR DESCRIPTION
**Pull request by R3vlyn**

### CODE REVIEW

**Comment by R3vlyn** [comment]
Example calls:
get localhost:3000/api/blocks
get localhost:3000/api/transactions
get localhost:3000/api/transaction/tito

Post calls are only for testing purposes due to the fact that adding will be done from the backup-validator.

File: models/block.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var mongoose = require('mongoose');
```
Comment (Merikw) [inline models/block.js:1]: Change name of file (blockSchema or whatever) and use block model from node module in the schema

File: models/block.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//var BlockHeader = require('./blockheader');
+//var Transaction = require('./transaction');
+var BlockSchema = new mongoose.Schema({
+    transactions: [{}],
```
Comment (Merikw) [inline models/block.js:5]: Is there a difference between coin and stake transactions?

File: models/blockheader.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+var mongoose = require('mongoose');
```
Comment (Merikw) [inline models/blockheader.js:1]: Change name of file (blockSchema or whatever) and use block model from node module in the schema

File: models/block.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    blockheader: {
+        hash: String,
+        previoushash: String,
+        validator: String,
+        timestamp: String,
+        reward: String,
```
Comment (Merikw) [inline models/block.js:11]: Shouldn't this be a Number?

File: models/block.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+var mongoose = require('mongoose');
+//var BlockHeader = require('./blockheader');
+//var Transaction = require('./transaction');
+var BlockSchema = new mongoose.Schema({
+    transactions: [{}],
+    blockheader: {
+        hash: String,
+        previoushash: String,
+        validator: String,
+        timestamp: String,
```
Comment (Merikw) [inline models/block.js:10]: Shouldn't this be a Date?

File: models/blockheader.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+var BlockHeaderSchema = new mongoose.Schema({
+    hash: String,
+    previoushash: String,
+    validator: String,
+    timestamp: String,
```
Comment (Merikw) [inline models/blockheader.js:6]: Shouldn't this be a Date?

File: models/blockheader.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    reward: String,
```
Comment (Merikw) [inline models/blockheader.js:7]: Shouldn't this be a Number?

File: models/transaction.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+var mongoose = require('mongoose');
+var TransactionSchema = new mongoose.Schema({
+    sender: String,
+    recipient: String,
+    amount: String,
```
Comment (Merikw) [inline models/transaction.js:5]: Shouldn't this be a Number?

File: models/transaction.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+    type: String,
+    timestamp: String
```
Comment (Merikw) [inline models/transaction.js:7]: Shouldn't this be a Date?

**Review by Merikw — COMMENTED** [review]
If changed / looked at the comments than it can be merged to dev (remove node modules from git I suppose?)

**Pull request merged**